web/src/SidebarTriggerButton.tsx

```diff
@@ -34,23 +35,28 @@ export default class SidebarTriggerButton extends PureComponent<
 
     // isReady (i.e. trigger button will appear) if:
     // 1. resource not currently building, AND
-    // 2. resource has built at least once (i.e. we're not waiting for the first build), AND
+    // 2. resource is not queued to build, AND
+    // 3. resource has built at least once (i.e. we're not waiting for the first build), AND
     //    ^ this will need to change with TRIGGER_MODE_MANUAL_NO_INITIAL
-    // 3. resource doesn't have a pending build (i.e. no pending changes, OR pending changes but it's a
-    //    manual resource)
-    // TODO: don't show trigger button if a manual resource has been queued for build (currently
-    //   have no way to detect this)
+    // 4. resource doesn't have a pending auto-build (i.e. no pending changes, OR pending
+    //    changes but it's a manual resource)
     let isReady =
+      !props.isQueued &&
       !props.isBuilding &&
       props.hasBuilt &&
       (!props.hasPendingChanges || isManualTriggerMode)
     let isDirty = props.hasPendingChanges && isManualTriggerMode
 
     return (
       <button
-        onClick={() => triggerUpdate(props.resourceName)}
+        onClick={(e: React.MouseEvent) => {
+          e.preventDefault() // don't select the resource in the sidebar
```

Comment:
if you want to prevent the event from bubbling up to the resource handler, use e.stopPropagation()

preventDefault seems wrong here - it's for preventing the default browser ui interactions from kicking in, which is not what we want

Reply:
here's a pretty good explanation
https://medium.com/@jacobwarduk/how-to-correctly-use-preventdefault-stoppropagation-or-return-false-on-events-6c4e3f31aedb

Reply:
also already tried that, I think there's some wonky behavior with the react `Link` component? Or else I'm missing something dumb, but here are my attempts to structure the DOM appropriately, and the weird shit they each do:
![Screen Shot 2019-11-05 at 2 20 55 PM](https://user-images.githubusercontent.com/6332648/68238878-f2ea3e80-ffd7-11e9-88dc-3eeb6c1d8872.png)
![Screen Shot 2019-11-05 at 2 20 20 PM](https://user-images.githubusercontent.com/6332648/68238879-f2ea3e80-ffd7-11e9-91bb-b33f2a804d07.png)

Reply:
sent https://github.com/windmilleng/tilt/pull/2497 to move the button out of the link.

also created a storybook so that you can iterate on this outside of tilt